### PR TITLE
BI-1721 - Show All - Breeding Methods table

### PR DESCRIPTION
**Pull request by timparsons**
# Description
**Story:** https://breedinginsight.atlassian.net/browse/BI-1721

Original bug was that the "Show All" button wasn't working for the Breeding Methods table.  When investigating how to fix this, there was a lot of duplicate and confusing code around pagination, so I refactored that to utilize `PaginationController` and the mechanism for a parent component to update page information as needed, and for the `ExpandableTable` component to now be primarily responsible for handling paging events from the `PaginationControls`.

# Dependencies
bi-api - [bug/BI-1721](https://github.com/Breeding-Insight/bi-api/pull/242)

# Testing
1. On a given table, click the 'Show All' button, and ensure that all records are now visible
2. Click the 'Show All' button again, and ensure that the table is paginated again
3. Ensure pagination works

Testing applies to the all tables:
- Admin Users
- Admin Programs
- Program Users
- Program Locations
- Program Breeding Methods
- Enabling System Breeding Methods
- All Germplasm
- Germplasm Lists
- Germplasm List Details
- Ontology (active)
- Ontology (archived)
- Experiments
- Import Experiment
- Import Ontology
- Import Germplasm

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [x] I have create/modified unit tests to cover this change
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to documentation
- [x] I have run TAF: https://github.com/Breeding-Insight/taf/actions/runs/4297630354

### CODE REVIEW

**Review by davedrp — APPROVED** [review]

File: src/breeding-insight/model/BiResponse.ts [hunk 1]
```diff
@@ -36,10 +36,10 @@ export class Metadata {
...
-  currentPage: Number;
-  totalCount: Number;
-  pageSize: Number;
+  totalPages: number;
+  currentPage: number;
+  totalCount: number;
+  pageSize: number;
 
   constructor(paginationResult?: any) {
     if (paginationResult){
```
[the remark this answers is not in the document]
Comment (timparsons) [reply src/breeding-insight/model/BiResponse.ts:45]: This makes sense.  For this specific instance, I tried changing it to `object`, then had other compilation errors.  If you're ok with it, I'd rather not change this now.

File: src/breeding-insight/model/view_models/PaginationUtilities.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  // }
+
+  // static getPaginationSelections(currentPage: number, pageSize: number, showAll: boolean): PaginationQuery {
+  //   if (showAll) {
+  //     return new PaginationQuery(0, pageSize, true);
+  //   } else {
+  //     return new PaginationQuery(
+  //       currentPage, pageSize, false);
+  //   }
+  // }
```
[the remark this answers is not in the document]
Comment (timparsons) [reply src/breeding-insight/model/view_models/PaginationUtilities.ts]: Fixed

**Comment by timparsons** [comment]
> Both "Germplasm" and "Experiments&Observations" are stuck in a recursive loop that keeps making requests to the back end.

@dmeidlin Good catch! This has been fixed

File: src/breeding-insight/model/BiResponse.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmeidlin) [reply src/breeding-insight/model/BiResponse.ts:45]: Sure, that's fine.

**Review by dmeidlin — APPROVED** [review]

**Pull request merged**